bridge/test/token/madByte/burn.ts

```diff
@@ -0,0 +1,348 @@
+    expectedState = await getState(fixture);
+    [eths] = await callFunctionAndGetReturnValues(
+      fixture.madByte,
+      "burn",
+      user,
+      [burnQuantity, 0]
+    );
+    let roundedEths = format(eths);
+    expectedState.Balances.madByte.user -= burnQuantity;
+    expectedState.Balances.eth.user += roundedEths;
+    expectedState.Balances.madByte.poolBalance -= eths.toBigInt();
+    expectedState.Balances.madByte.totalSupply -= burnQuantity;
+    expectedState.Balances.eth.madByte -= eths.toBigInt();
+    expect(await getState(fixture)).to.be.deep.equal(expectedState);
+  });
+
+  it("Should fail to burn more than possible", async () => {
+    // Try to burn one more than minted
+    let burnQuantity = BigNumber.from(madBytes).add(1);
+    await expect(
+      fixture.madByte.connect(user).burn(burnQuantity, 0)
+    ).to.be.revertedWith(
+      "MadByte: The number of tokens to be burned is greater than the Total Supply!"
+    );
+  });
+
+  it("Should fail to burn 0 tokens", async () => {
+    // Try to burn 0
+    let burnQuantity = 0;
+    await fixture.madByte.connect(user).mint(minMadBytes, {
+      value: ethers.utils.parseEther(eth.toString()),
+    });
+    await expect(
+      fixture.madByte.connect(user).burn(burnQuantity, 0)
+    ).to.be.revertedWith(
+      "MadByte: The number of MadBytes to be burn should be greater than 0!"
+    );
+  });
+
+  it("Should burn to an address", async () => {
+    let burnQuantity = madBytes;
+    // Round eths to avoid consumed gas on state comparison
+    expectedState = await getState(fixture);
+    //Eths to receive
+    [eths] = await callFunctionAndGetReturnValues(
+      fixture.madByte,
+      "burnTo",
+      user,
+      [admin.address, burnQuantity, 0]
+    );
+    let roundedEths = format(eths);
+    expectedState.Balances.madByte.user -= burnQuantity.toBigInt();
+    expectedState.Balances.eth.admin += roundedEths;
+    expectedState.Balances.madByte.poolBalance -= eths.toBigInt();
+    expectedState.Balances.madByte.totalSupply -= burnQuantity.toBigInt();
+    expectedState.Balances.eth.madByte -= eths.toBigInt();
+    expect(await getState(fixture)).to.be.deep.equal(expectedState);
+  });
+
+  it("Should fail to burn to an address more than possible", async () => {
+    // Try to burn one more than minted
+    let burnQuantity = BigNumber.from(madBytes).add(1);
+    await expect(
+      fixture.madByte.connect(user).burnTo(admin.address, burnQuantity, 0)
+    ).to.be.revertedWith(
+      "MadByte: The number of tokens to be burned is greater than the Total Supply!"
+    );
+  });
+
+  it("Should fail to burn 0 tokens to an address", async () => {
+    let burnQuantity = 0;
+    await expect(
+      fixture.madByte.connect(user).burnTo(admin.address, burnQuantity, 0)
+    ).to.be.revertedWith(
+      "MadByte: The number of MadBytes to be burn should be greater than 0!"
+    );
+  });
+
+  it("Should fail to burn to 0 address", async () => {
```

Comment:
This test seems to be same in the line 169

Reply:
Your absolutely right. Fixed